### DISCUSSION TITLE
There is no plans about sampling strategy from metricsGenerator ?

### DISCUSSION THREAD

**Discussion by pingping95** [grafana/tempo#3740] [General]
Hi.

I am running LGTM Stack and using tempo well.

I have a question.

Is there no plan about sampling strategy from metricsGenerator?

Metrics Generator is very comfortable component to generate RED Metrics and serviceGraph Metrics.

When i use metricsGenerator in order to use Full RED Metrics with SLI & SLO, I have to sample all traces and can`t use sampling strategy.

(Actually, I need FULL RED Metrics and Do not need all traces data. and TRACE datas are often so huge.)

So, Is there any plan to implement this ?

thanks.

**Comment by yvrhdn** [comment]
We have two issues for this open right now:
- https://github.com/grafana/tempo/issues/1515
- https://github.com/grafana/tempo/issues/2684

In short: yes, we want the metrics-generator to correctly estimate metrics from sampled traces but we didn't get to implementing it yet.

I think we need some research into which attributes we can use (take a look at the OTel convention, what the OTel Collector and some other agents do) and then implement it in the different processors.
If you are interested in doing this kind of work let us know and we can help you out!
